src/Timeline/TimelineImageFactory.php

```diff
@@ -0,0 +1,172 @@
+<?php
+
+namespace AppBundle\Timeline;
+
+use AppBundle\Entity\Timeline\Measure;
+use AppBundle\Repository\Timeline\MeasureRepository;
+use Symfony\Component\Translation\TranslatorInterface;
+
+class TimelineImageFactory
+{
+    private const OUTPUT_IMAGE_NAME = 'timeline-image.jpg';
+
+    private const TEXT_FONT_BOLD = 'Roboto-Bold.ttf';
+    private const TEXT_FONT_REGULAR = 'Roboto-Regular.ttf';
+
+    private const TEXT_SIZE_BOLD = 60;
+    private const TEXT_SIZE_REGULAR = 40;
+
+    private const TEXTS_COORDINATES = [
+        Measure::STATUS_DONE => ['x' => 760, 'y' => 674],
+        Measure::STATUS_IN_PROGRESS => ['x' => 760, 'y' => 772],
+        Measure::STATUS_UPCOMING => ['x' => 760, 'y' => 870],
+    ];
+
+    private const CHART = [
+        'center' => ['x' => 378, 'y' => 745],
+        'innerRadius' => 151,
+        'outerRadius' => 388,
+    ];
+
+    private $measureRepository;
+    private $translator;
+    private $cacheDirectory;
+    private $webDirectory;
+
+    public function __construct(
+        MeasureRepository $measureRepository,
+        TranslatorInterface $translator,
+        string $cacheDirectory,
+        string $webDirectory
+    ) {
+        $this->measureRepository = $measureRepository;
+        $this->translator = $translator;
+        $this->cacheDirectory = $cacheDirectory;
+        $this->webDirectory = $webDirectory;
+    }
+
+    public function createImage(string $locale): string
+    {
+        $counts = $this->getMeasureCounts();
+
+        $image = $this->createImageFromPng($this->getImageTemplatePath($locale));
+
+        $this->drawTexts($image, $counts, $locale);
+        $this->drawChart($image, $counts);
+
+        $this->saveImage($image, $imagePath = $this->getOutputImagePath());
```

Comment:
No need to pass the second arg, inline the method call in `saveImage` and return the output path from there.